src/Stiefel.jl

```diff
@@ -0,0 +1,199 @@
+using LinearAlgebra: diag, qr, tr, svd, mul!, lyap
+
+@doc doc"""
+    Stiefel{m,n,T} <: Manifold
+
+The Stiefel manifold consists of all $m\times n$, $m\geq n$ orthonormal matrices, i.e.
+
+````math
+\mathcal M = \{ x \in \mathbb K^{m\times n} : \overline{x}^{\mathrm{T}}x = I_n \}
+````
+where $\mathbb K \in \{\mathbb R, \mathbb C\}$ and
+$I_n \in \mathbb R^{n\times n}$ denotes the unit matrix.
+
+The tangent space at a point $x\in\mathcal M$ is given by
+````math
+T_x\mathcal M = \{ v \in \mathbb{K}^{m\times n} : x^{\mathrm{T}}v + v^{\mathrm{T}}x=0_n\},
+````
+where $0_n$ is the $n\times n$ zero matrix.
+
+# Constructor
+    Stiefel(m,n,T=Real)
+
+generate the (real-valued) Stiefel manifold of $m\times n$ dimensional orthonormal matrices.
+"""
+struct Stiefel{M,N,T} <: Manifold end
+Stiefel(m::Int, n::Int,T::Type = Type{Real} ) = Stiefel{m,n,T}()
```

Comment:
My comment for constructor of `FixedRankMatrices` is also relevant here (`Real` vs `Type{Real}`)